src/wp-includes/media.php

```diff
@@ -5564,6 +5676,135 @@ function wp_get_loading_attr_default( $context ) {
 	return 'lazy';
 }
 
+/**
+ * Get loading optimization attributes.
+ *
+ * @since 6.3.0
+ *
+ * @param string $tag_name The tag name.
+ * @param array[] $attr Array of the attributes for the tag.
+ * @param string $context Context for the element for which the loading optimization attribute is requested.
+ *
+ * @return array[] loading optimization attributes.
+ */
+function wp_get_loading_optimization_attributes( $tag_name, $attr, $context ) {
+	global $wp_query;
+
+	$loading_attrs = array();
+
+	// All three parameters are required.
+	if ( empty( $tag_name ) || empty( $attr ) || empty( $context ) ) {
+		return $loading_attrs;
+	}
+
+	// Image tag must have a minimum of width and height.
+	if ( 'img' === $tag_name && ( ! isset( $attr['width'] ) || ! isset( $attr['height'] ) ) ) {
+		return $loading_attrs;
+	}
+
+	// Skip lazy-loading for the overall block template, as it is handled more granularly.
+	// The skip is also applicable for fetchpriority.
+	if ( 'template' === $context ) {
+		return $loading_attrs;
```

Comment:
The skip condition was originally present. I assumed that the condition stands valid for `fetchpriority` too. 
Need you thoughts on this @felixarntz @joemcgill.